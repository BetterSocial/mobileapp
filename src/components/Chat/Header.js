import * as React from 'react';
import IconEP from 'react-native-vector-icons/Entypo';
import {ChannelAvatar} from 'stream-chat-react-native';
import {Image, StyleSheet, Text, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import AnonymousIcon from '../../screens/ChannelListScreen/elements/components/AnonymousIcon';
import DefaultGroupProfilePicture from '../../assets/images/default-group-picture.png';
import GlobalButton from '../Button/GlobalButton';
import MemoIc_arrow_back_white from '../../assets/arrow/Ic_arrow_back_white';
import {CHANNEL_TYPE_ANONYMOUS} from '../../utils/constants';
import {Context} from '../../context';
import {fonts} from '../../utils/fonts';
import {getChatName, getGroupMemberCount} from '../../utils/string/StringUtils';
import {COLORS} from '../../utils/theme';

const Header = ({onBack}) => {
  const navigation = useNavigation();
  const [channelClient] = React.useContext(Context).channel;
  const [profileContext] = React.useContext(Context).profile;
  const {channel} = channelClient;
  const [chatName, setChatName] = React.useState(null);
  let username = channelClient.channel?.data?.name;
  const channelType = channelClient?.channel?.data?.channel_type;
  if (channelType === CHANNEL_TYPE_ANONYMOUS) {
    username = `Anonymous ${channelClient?.channel?.data?.anon_user_info_emoji_name}`;
  }

  React.useEffect(() => {
    const unsubscribe = navigation.addListener('focus', () => {
      if (channelClient.channel?.data?.name) {
        setChatName(
          getChatName(channelClient?.channel?.data?.name, profileContext?.myProfile?.username)
        );
      }
    });

    return unsubscribe;
  }, [navigation]);

  React.useEffect(() => {
    if (
      profileContext.myProfile &&
      typeof profileContext.myProfile === 'object' &&
      typeof username === 'string'
    ) {
      setChatName(getChatName(username, profileContext.myProfile.username));
    }
  }, [JSON.stringify(profileContext.myProfile), username]);

  const renderHeaderImage = () => {
    if (channelType === CHANNEL_TYPE_ANONYMOUS) {
      return (
        <View style={{marginLeft: 18}}>
          <AnonymousIcon
            size={42}
            color={channel?.data?.anon_user_info_color_code}
            emojiCode={channel?.data?.anon_user_info_emoji_code}
          />
        </View>
      );
    }

    if (channel?.data?.image) {
      if (channel?.data?.image.indexOf('res.cloudinary.com') > -1) {
        return <Image source={{uri: channel?.data?.image}} style={styles.image} />;
      }
      return (
        <Image
          source={{uri: `data:image/jpg;base64,${channel?.data?.image}`}}
          style={styles.image}
        />
      );
    }
    if (getGroupMemberCount(channel) > 2) {
      return <Image source={DefaultGroupProfilePicture} style={styles.image} />;
    }
    return (
      <View style={styles.containerAvatar}>
        <ChannelAvatar size={42} channel={channel} />
      </View>
    );
  };

  const handleBackBtn = () => {
    if (onBack && typeof onBack === 'function') {
      return onBack();
    }
    return navigation.goBack();
  };

  return (
    <View style={styles.container}>
      <View style={[styles.row, {flex: 1}]}>
        <GlobalButton buttonStyle={styles.backContainer} onPress={handleBackBtn}>
          <MemoIc_arrow_back_white width={20} height={12} />
        </GlobalButton>
        <GlobalButton
          buttonStyle={styles.backContainer}
          onPress={() => navigation.push('GroupInfo')}>
          <View style={styles.touchable}>
            {renderHeaderImage()}
            <Text numberOfLines={1} style={styles.name}>
              {chatName}{' '}
            </Text>
          </View>
        </GlobalButton>
      </View>
      <View style={styles.row}>
        <GlobalButton
          style={styles.btnOptions}
          onPress={() =>
            navigation.push('GroupSetting', {
              username,
              focusChatName: true
            })
          }>
          <IconEP name="dots-three-vertical" size={12.87} color={COLORS.almostBlack} />
        </GlobalButton>
      </View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    height: 50,
    backgroundColor: COLORS.anon_primary,
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 22,
    justifyContent: 'space-between'
  },
  containerAvatar: {
    marginLeft: 18
  },
  btnOptions: {
    paddingLeft: 0,
    paddingRight: 0
  },
  btnSearch: {
    paddingRight: 9.165
  },

  row: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  image: {
    height: 40,
    width: 40,
    borderRadius: 40 / 2,
    marginLeft: 18
  },
  name: {
    marginLeft: 11,
    fontFamily: fonts.inter[600],
    color: COLORS.almostBlack,
    fontSize: 14,
    width: '70%'
  },
  touchable: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1
  },
  backContainer: {
    paddingLeft: 0
  }
});
