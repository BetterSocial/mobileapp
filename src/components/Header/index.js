import React from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableNativeFeedback,
  View,
} from 'react-native';
import ArrowLeftIcon from '../../../assets/icons/arrow-left.svg';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
const Header = ({title, onPress, titleStyle = {}, containerStyle={}}) => {
  const renderHeader = () => {
    if (Platform.OS === 'android') {
      return (
        <TouchableNativeFeedback onPress={onPress}>
          <View style={{padding : 10, marginLeft : -4}}>
            <ArrowLeftIcon width={20} height={12} fill="#000"/>
          </View>
        </TouchableNativeFeedback>
      );
    } else {
      return (
        <TouchableHighlight onPress={onPress}>
          <View style={{padding : 10, marginLeft : -8}}>
            <ArrowLeftIcon width={20} height={12} fill="#000"/>
          </View>
        </TouchableHighlight>
      );
    }
  };
  return (
    <View style={{...styles.container, ...containerStyle}}>
      {renderHeader()}
      <Text style={{...styles.text, ...titleStyle}}>{title}</Text>
      <View style={{width: 20, height: 12}} />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  text: {
    color: colors.black,
    fontFamily: fonts.poppins[600],
    fontSize: 14,
    marginLeft : -20,
    fontWeight: 'bold',
  },
});
