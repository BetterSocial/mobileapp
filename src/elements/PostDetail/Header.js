import * as React from 'react';
import {StyleSheet, Text, View, Image, TouchableOpacity} from 'react-native';

import Icon from 'react-native-vector-icons/Octicons';

import {fonts} from '../../utils/fonts';
import Gap from '../../components/Gap';

const Header = ({onSetting, onSearch}) => {
  return (
    <View style={styles.header}>
      <View style={{flexDirection: 'row'}}>
        <Image
          source={require('../../assets/images/ProfileDefault.png')}
          style={styles.image}
        />
        <TouchableOpacity style={styles.setting} onPress={onSearch}>
          <Icon name="search" size={20} />
        </TouchableOpacity>
      </View>
      <Text style={styles.headerTitle}>Post</Text>
      <View style={{flexDirection: 'row'}}>
        <Gap style={{width: 50}} />
        <TouchableOpacity style={styles.setting} onPress={onSetting}>
          <Icon name="settings" size={20} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  image: {
    height: 32,
    width: 32,
    marginRight: 10,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 20,
    borderBottomColor: '#e8e8e8',
    borderBottomWidth: 1,
    paddingVertical: 10,
  },
  setting: {
    backgroundColor: '#e8e8e8',
    width: 32,
    height: 32,
    borderRadius: 32 / 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerTitle: {
    fontFamily: fonts.inter[700],
    fontSize: 18,
    color: '#363131',
  },
});
