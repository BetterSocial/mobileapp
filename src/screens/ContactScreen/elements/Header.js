import * as React from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import PropTypes from 'prop-types';
import ArrowLeftIcon from '../../../../assets/icons/arrow-left.svg';
import GlobalButton from '../../../components/Button/GlobalButton';
import {COLORS, SIZES} from '../../../utils/theme';
import dimen from '../../../utils/dimen';

const Header = ({
  title,
  subTitle,
  onPress,
  titleStyle = {},
  containerStyle = {},
  onPressSub,
  disabledNextBtn
}) => {
  const renderHeader = () => {
    if (Platform.OS === 'android') {
      return (
        <GlobalButton buttonStyle={styles.backContainer} onPress={onPress}>
          <View style={styles.content(-4)}>
            <ArrowLeftIcon
              width={dimen?.normalizeDimen(20)}
              height={dimen?.normalizeDimen(20)}
              fill="#000"
            />
          </View>
        </GlobalButton>
      );
    }
    return (
      <GlobalButton buttonStyle={styles.backContainer} onPress={onPress}>
        <View style={styles.content(-8)}>
          <ArrowLeftIcon
            width={dimen?.normalizeDimen(20)}
            height={dimen?.normalizeDimen(20)}
            fill="#000"
          />
        </View>
      </GlobalButton>
    );
  };
  return (
    <View style={{...styles.container, ...containerStyle}}>
      {renderHeader()}
      <View style={styles.containerTitle}>
        <Text style={{...styles.text, ...titleStyle}}>{title}</Text>
      </View>
      <GlobalButton
        disabled={disabledNextBtn}
        buttonStyle={styles.nextContainer}
        onPress={onPressSub}>
        <Text style={[styles.text, {color: disabledNextBtn ? COLORS.gray6 : COLORS.anon_primary}]}>
          {subTitle}
        </Text>
      </GlobalButton>
    </View>
  );
};

export default Header;

Header.propTypes = {
  onPressSub: PropTypes.func,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  onPress: PropTypes.func,
  titleStyle: PropTypes.object,
  containerStyle: PropTypes.object,

  disabledNextBtn: PropTypes.bool
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: SIZES.base,
    alignItems: 'center'
  },
  content: (marginLeft) => ({
    padding: 10,
    marginLeft
  }),
  text: {
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: 16,
    color: COLORS.black,
    lineHeight: 20,
    alignSelf: 'center'
  },
  gap: {width: 20, height: 12},
  containerTitle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  backContainer: {
    paddingLeft: 0
  },
  nextContainer: {
    paddingLeft: 0
  }
});
