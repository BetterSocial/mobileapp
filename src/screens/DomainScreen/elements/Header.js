import * as React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  StyleSheet,
  Linking,
} from 'react-native';
import Tooltip from 'react-native-walkthrough-tooltip';
import SimpleToast from 'react-native-simple-toast';

import MemoIc_interface from '../../../assets/icons/Ic_interface';
import MemoIc_question_mark from '../../../assets/icons/Ic_question_mark';
import MemoIc_rectangle_gradient from '../../../assets/Ic_rectangle_gradient';
import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
import {SIZES, COLORS} from '../../../utils/theme';
import {SingleSidedShadowBox, Gap} from '../../../components';
import StringConstant from '../../../utils/string/StringConstant';
import {Context} from '../../../context';
import {
  followDomain,
  getDomainIdIFollow,
  unfollowDomain,
} from '../../../service/domain';
import {addIFollowByID, setIFollow} from '../../../context/actions/news';
import {TouchableNativeFeedback} from 'react-native';
import {colors} from '../../../utils/colors';
import MemoDomainProfilePicture from '../../../assets/icon/DomainProfilePictureEmptyState';

const lorem =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent placerat erat tellus, non consequat mi sollicitudin quis.';

const Header = ({image, domain, description, followers, onPress, iddomain}) => {
  let [isTooltipShown, setIsTooltipShown] = React.useState(false);
  const [follow, setFollow] = React.useState(false);
  const [news, dispatch] = React.useContext(Context).news;
  const [dataFollow] = React.useState({
    domainId: iddomain,
    source: 'domain_page',
  });
  let {ifollow} = news;

  const openDomainLink = async () => {
    let isURL = await Linking.canOpenURL(`https://${domain}`);
    console.log(isURL);
    if (isURL) {
      Linking.openURL(`https://${domain}`);
    } else {
      SimpleToast.show(StringConstant.domainCannotOpenURL, SimpleToast.SHORT);
    }
  };
  React.useEffect(() => {
    getIFollow();
  }, [iddomain, ifollow]);
  const getIFollow = async () => {
    if (ifollow.length === 0) {
      let res = await getDomainIdIFollow();
      setIFollow(res.data, dispatch);
    } else {
      setFollow(JSON.stringify(ifollow).includes(iddomain));
    }
  };
  const handleFollow = async () => {
    setFollow(true);
    const res = await followDomain(dataFollow);
    if (res.code === 200) {
      addIFollowByID(
        {
          domain_id_followed: iddomain,
        },
        dispatch,
      );
    } else {
      console.log('error follow domain');
    }
  };
  const handleUnFollow = async () => {
    setFollow(false);
    const res = await unfollowDomain(dataFollow);
    if (res.code === 200) {
      let newListFollow = await ifollow.filter(function (obj) {
        return obj.domain_id_followed !== iddomain;
      });
      setIFollow(newListFollow, dispatch);
    } else {
      console.log('error unfollow domain');
    }
  };

  return (
    <View style={styles.headerDomain}>
      <View style={styles.row}>
        <View style={styles.wrapperImage}>
          {image ? (
            <>
              <View style={styles.circleImageWrapper} />
              <Image
                source={{uri: image}}
                style={[styles.circleImage, StyleSheet.absoluteFillObject]}
              />
            </>
          ) : (
            <MemoDomainProfilePicture />
          )}
        </View>
        <View style={styles.wrapperHeader}>
          {follow ? (
            <TouchableNativeFeedback onPress={() => handleUnFollow()}>
              <View style={styles.buttonFollowing}>
                <Text style={styles.textButtonFollowing}>Following</Text>
              </View>
            </TouchableNativeFeedback>
          ) : (
            <TouchableNativeFeedback onPress={() => handleFollow()}>
              <View style={styles.buttonFollow}>
                <Text style={styles.textButtonFollow}>Follow</Text>
              </View>
            </TouchableNativeFeedback>
          )}
          <Gap width={normalize(SIZES.base)} />
          <TouchableOpacity
            style={styles.buttonBlock}
            onPress={() => onPress()}>
            <Text style={styles.blockButtonText}>Block</Text>
          </TouchableOpacity>
        </View>
      </View>
      <Gap height={normalize(12)} />
      <View style={styles.row}>
        <Text style={styles.domainName}>{domain}</Text>
        <View style={{justifyContent: 'center'}}>
          <TouchableOpacity
            style={styles.openInBrowserIcon}
            onPress={openDomainLink}>
            <MemoIc_interface width={17} height={17} />
          </TouchableOpacity>
        </View>
      </View>
      <Gap height={normalize(2)} />
      <View style={[styles.row, {alignItems: 'center'}]}>
        <Text style={styles.followersNumber}>{followers}k</Text>
        <Gap width={normalize(4)} />
        <Text style={styles.followersText}>Followers</Text>
      </View>
      <Gap height={normalize(14)} />
      <Text style={styles.domainDescription}>
        {description ? description : lorem}
      </Text>
      <Gap height={normalize(10)} />
      <View style={{flexDirection: 'row', alignItems: 'center', flex: 1}}>
        <View style={{flex: 1, paddingBottom: 0}}>
          <MemoIc_rectangle_gradient width={'100%'} height={normalize(18)} />
        </View>
        <Tooltip
          // allowChildInteraction={false}
          isVisible={isTooltipShown}
          placement={'bottom'}
          backgroundColor={'rgba(0,0,0,0)'}
          showChildInTooltip={false}
          onClose={() => setIsTooltipShown(false)}
          contentStyle={styles.tooltipShadowContainer}
          arrowSize={{width: 0, height: 0}}
          content={
            <View>
              <Text style={styles.tooltipContent}>
                {description ? description : `${lorem} ${lorem} ${lorem}`}
              </Text>
            </View>
          }>
          <TouchableOpacity
            onPress={() => setIsTooltipShown(true)}
            style={{
              padding: 8,
              paddingBottom: 8,
              paddingTop: 8,
              paddingRight: 12,
            }}>
            <MemoIc_question_mark
              width={normalize(17)}
              height={normalize(17)}
            />
          </TouchableOpacity>
        </Tooltip>
      </View>
      <Gap height={normalize(16)} />
    </View>
  );
};

const styles = StyleSheet.create({
  icon: {flexDirection: 'row', alignItems: 'center'},
  desc: {
    fontSize: normalizeFontSize(14),
    fontFamily: fonts.inter[400],
    lineHeight: normalizeFontSize(16),
  },
  containerFollowers: {flexDirection: 'row'},
  followers: {
    color: '#00ADB5',
    fontFamily: fonts.inter[400],
    fontSize: normalizeFontSize(16),
    fontWeight: '700',
  },
  wrapperDomain: {flexDirection: 'row', marginTop: 8},
  iconDomain: {marginStart: 8, justifyContent: 'center'},
  domain: {
    fontSize: normalizeFontSize(24),
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
  },
  actionText: (color) => ({
    fontSize: normalizeFontSize(14),
    color,
  }),
  headerDomain: {
    flexDirection: 'column',
    backgroundColor: 'white',
    paddingHorizontal: 20,
    borderTopColor: 'transparent',
    // shadowColor: '#000',
    // shadowOffset: {width: 0, height: 1},
    // shadowOpacity: 0.8,
    // shadowRadius: 1,
    paddingTop: 15,
  },
  container: {
    flexDirection: 'row',
  },
  image: {height: '100%', width: '100%', borderRadius: 45},
  containerImage: {flex: 1.3},
  boxImage: {
    borderRadius: 45,
    borderWidth: 0.2,
    borderColor: 'rgba(0,0,0,0.5)',
    width: 60,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonPrimary: {
    height: normalize(36),
    backgroundColor: '#00ADB5',
    // flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
  },
  buttonBlock: {
    // flex: 1,
    height: normalize(36),
    width: normalize(88),
    borderWidth: 0.5,
    borderRadius: 8,
    borderColor: '#FF2E63',
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  blockButtonText: {
    fontSize: normalizeFontSize(12),
    color: '#FF2E63',
    paddingHorizontal: 25,
  },
  wrapperHeader: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    flexDirection: 'row',
  },
  width: (wid) => ({
    width: wid,
  }),
  height: (height) => ({
    height,
  }),
  wrapperImage: {
    borderRadius: normalize(50),
    borderColor: 'rgba(0,0,0,0.5)',
    width: normalize(100),
    height: normalize(100),
    justifyContent: 'center',
    alignItems: 'center',
  },
  row: {
    flexDirection: 'row',
  },
  followButtonText: {
    fontSize: normalizeFontSize(14),
    color: 'white',
    paddingHorizontal: 25,
  },
  domainName: {
    fontSize: normalizeFontSize(24),
    fontFamily: fonts.inter[500],
    lineHeight: normalizeFontSize(29),
    color: '#000000',
  },
  followersNumber: {
    color: '#00ADB5',
    fontFamily: fonts.inter[700],
    fontSize: normalizeFontSize(14),
    lineHeight: normalizeFontSize(17),
  },
  followersText: {
    color: COLORS.black,
    fontFamily: fonts.inter[400],
    fontSize: normalizeFontSize(14),
    lineHeight: normalizeFontSize(17),
  },
  domainDescription: {
    fontFamily: fonts.inter[400],
    lineHeight: normalizeFontSize(17),
    fontSize: normalizeFontSize(14),
    // lineHeight: 17,
  },
  shadowBox: {
    paddingBottom: 8,
  },
  tooltipContent: {
    fontFamily: fonts.inter[400],
    fontSize: normalizeFontSize(14),
    lineHeight: normalizeFontSize(17),
    color: COLORS.blackgrey,
  },
  tooltipShadowContainer: {
    paddingHorizontal: 10,
    shadowColor: '#000',
    shadowOffset: {
      width: 2,
      height: 2,
    },
    shadowOpacity: 1,
    shadowRadius: 3.84,
    elevation: 5,
  },
  arrow: {
    position: 'absolute',
    width: 10,
    height: 10,
    backgroundColor: 'white',
    top: 23,
    zIndex: 10000000,
    left: 10,
    transform: [{rotate: '45deg'}],
    shadowColor: '#000',
    shadowOffset: {
      width: 2,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 5,
  },
  openInBrowserIcon: {
    padding: normalize(4),
    paddingHorizontal: normalize(12),
    top: 0,
  },
  buttonFollowing: {
    width: normalize(88),
    height: normalize(36),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: colors.bondi_blue,
    borderRadius: 8,
  },
  textButtonFollowing: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: normalizeFontSize(12),
    lineHeight: normalizeFontSize(24),
    color: colors.bondi_blue,
  },
  buttonFollow: {
    width: normalize(88),
    height: normalize(36),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    backgroundColor: colors.bondi_blue,
    color: colors.white,
  },
  textButtonFollow: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: normalizeFontSize(12),
    lineHeight: normalizeFontSize(24),
    color: colors.white,
  },
  circleImage: {
    height: normalize(100),
    width: normalize(100),
    alignSelf: 'center',
    borderRadius: normalize(100),
    resizeMode: 'cover',
    zIndex: -10000,
  },
  circleImageWrapper: {
    height: normalize(100),
    width: normalize(100),
    borderRadius: normalize(50),
    borderWidth: 0.2,
    borderColor: 'rgba(0,0,0,0.5)',
  },
});

export default Header;
