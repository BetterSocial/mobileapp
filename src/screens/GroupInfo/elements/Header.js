import * as React from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import {Gap} from '../../../components';
import MemoIc_arrow_back from '../../../assets/arrow/Ic_arrow_back';
import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
import {trimString} from '../../../utils/string/TrimString';
import {COLORS} from '../../../utils/theme';

const Header = ({title}) => {
  const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <TouchableOpacity testID="onBack" onPress={() => navigation.goBack()}>
        <MemoIc_arrow_back width={20} height={12} />
      </TouchableOpacity>
      <Text testID="title" style={styles.title}>
        {trimString(title, 21)}
      </Text>
      <Gap width={20} />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    height: normalize(48),
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 12
  },
  title: {
    fontSize: normalizeFontSize(16),
    fontFamily: fonts.inter[600],
    lineHeight: normalizeFontSize(19.36),
    color: COLORS.black
  }
});
