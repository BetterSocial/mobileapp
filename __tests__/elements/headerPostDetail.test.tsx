import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../../src/elements/PostDetail/Header';

configure({adapter: new Adapter()});

describe('component Header', () => {
  it('Header snapshot', () => {
    const component = shallow(
      <Header onSearch={() => {}} onSetting={() => {}} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Header renders correctly', () => {
    const tree = renderer
      .create(<Header onSearch={() => {}} onSetting={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
