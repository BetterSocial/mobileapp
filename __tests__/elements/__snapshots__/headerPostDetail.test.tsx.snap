// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Header Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#e8e8e8",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/ProfileDefault.png",
        }
      }
      style={
        Object {
          "height": 32,
          "marginRight": 10,
          "width": 32,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e8e8e8",
          "borderRadius": 16,
          "height": 32,
          "justifyContent": "center",
          "opacity": 1,
          "width": 32,
        }
      }
    >
      <Icon
        name="search"
        size={20}
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#363131",
        "fontFamily": "Inter-Bold",
        "fontSize": 18,
      }
    }
  >
    Post
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e8e8e8",
          "borderRadius": 16,
          "height": 32,
          "justifyContent": "center",
          "opacity": 1,
          "width": 32,
        }
      }
    >
      <Icon
        name="settings"
        size={20}
      />
    </View>
  </View>
</View>
`;

exports[`component Header Header snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#e8e8e8",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/ProfileDefault.png",
        }
      }
      style={
        Object {
          "height": 32,
          "marginRight": 10,
          "width": 32,
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e8e8e8",
          "borderRadius": 16,
          "height": 32,
          "justifyContent": "center",
          "width": 32,
        }
      }
    >
      <Icon
        name="search"
        size={20}
      />
    </ForwardRef>
  </View>
  <Text
    style={
      Object {
        "color": "#363131",
        "fontFamily": "Inter-Bold",
        "fontSize": 18,
      }
    }
  >
    Post
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Gap
      style={
        Object {
          "width": 50,
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e8e8e8",
          "borderRadius": 16,
          "height": 32,
          "justifyContent": "center",
          "width": 32,
        }
      }
    >
      <Icon
        name="settings"
        size={20}
      />
    </ForwardRef>
  </View>
</View>
`;
